### PR TITLE
zh translation: docs/setup/additional setup/sidecar injection/index.md

### PR DESCRIPTION
**Pull request by orangegzx**
[ ] Configuration Infrastructure
[x] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

### CODE REVIEW

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 1]
```diff
@@ -13,22 +13,25 @@ test: no
 
 ## 注入{#injection}
 
-为了充分利用 Istio 的所有特性，网格中的 pod 必须运行一个 Istio sidecar 代理。
+为了充分利用 Istio 的所有特性，网格中的 pod 必须运行一个 Istio Sidecar 代理。
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:16]:
```suggestion
为了充分利用 Istio 的所有特性，网格中的 Pod 必须运行一个 Istio Sidecar 代理。
```

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 2]
```diff
@@ -13,22 +13,25 @@ test: no
...
 
-下面的章节描述了向 pod 中注入 Istio sidecar 的两种方法：使用 [`istioctl`](/zh/docs/reference/commands/istioctl) 手动注入或启用 pod 所属命名空间的 Istio sidecar 注入器自动注入。
+下面的章节描述了向 pod 中注入 Istio Sidecar 的两种方法：使用 [`istioctl`](/zh/docs/reference/commands/istioctl) 手动注入或启用 Pod 所属命名空间的 Istio sidecar 注入器自动注入。
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:18]:
```suggestion
下面的章节描述了向 Pod 中注入 Istio Sidecar 的两种方法：使用 [`istioctl`](/zh/docs/reference/commands/istioctl) 手动注入或启用 Pod 所属命名空间的 Istio Sidecar 注入器自动注入。
```

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 3]
```diff
@@ -48,249 +51,186 @@ $ istioctl kube-inject \
...
-
-Sidecar 注入 webhook 是默认启用的。如果你希望禁用 webhook，可以使用 [Helm](/zh/docs/setup/install/helm/) 将 `sidecarInjectorWebhook.enabled` 设置为 `false`。
-
-还有很多[其他选项](/zh/docs/reference/config/installation-options/#sidecar-injector-webhook-options)可以配置。
+请注意，区别于手动注入，自动注入发生在 Pod 层面。你将看不到 Deployment 本身有任何更改。取而代之，需要检查单独的 Pod（使用 `kubectl describe`）来查询被注入的代理。
 
 #### 部署应用{#deploying-an-app}
 
-部署 sleep 应用。验证 deployment 和 pod 只有一个容器。
+部署 Sleep 应用。验证 Deployment 和 Pod 只有一个容器。
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:81]: sleep 是固定的，可以不用大写。
```suggestion
部署 sleep 应用。验证 Deployment 和 Pod 只有一个容器。
```

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 4]
```diff
@@ -48,249 +51,186 @@ $ istioctl kube-inject \
...
+1. 如果两个标签都没有设置，如果启用 `.values.sidecarInjectorWebhook.enableNamespacesByDefault`， 则会注入 Pod。在默认情况下是不启用的，所以 Pod 通常不会被注入。
 
-##### _**策略**_{#policy}
+## 自定义注入{#customizing-injection}
 
-`disabled` - 默认情况下不会将 sidecar 注入到 pod 中。在 pod 模板规范中添加 `sidecar.istio.io/inject` 的值为 true 来覆盖默认值并启用注入。
+通常，Pod 的注入是基于 Sidecar 注入模板，在 `istio-sidecar-injector` Configmap 中配置。每个 Pod 的配置可用于覆盖各个 Pod 上的选项。可通过在 Pod 中添加一个 `istio-proxy` Container来完成。Sidecar 注入将会把自定义的任何配置视为默认注入模板的覆盖。
 
-`enabled` - Sidecar 将默认注入到 pod 中。在 pod 模板规范中添加 `sidecar.istio.io/inject` 的值为 false 来覆盖默认值并禁用注入。
+自定义这些设置时，需格外小心，因为允许完全自定义生成的 Pod，包括进行一些更改而导致 Sidecar Container 无法正常运行。
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:172]: Container 翻译一下，容器。

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 5]
```diff
@@ -48,249 +51,186 @@ $ istioctl kube-inject \
...
+{{< /text >}}
+
+通常，可以设置 Pod 中的任何字段。但是必须注意某些字段：
+
+* Kubernetes 要求在注入运行之前配置 `image`。虽然可以您可以设置一个指定的 Image 来覆盖默认的 `image` 配置，但建议将 `image` 设置为 `auto`，可使 Sidecar 注入自动选择要使用的 Image。
+* `Pod` 中一些字段取决于相关设置。例如，CPU 请求必须小于 CPU 限制。如果两个字段没有一起配置， Pod 可能会无法启动。
+
+另外，某些字段可通过在 Pod 上的 [Annotations](/zh/docs/reference/config/annotations/) 配置，但是不建议使用上述方法进行自定义设置。
+
+### 自定义模板[试行]{custom-templates-[experimental]}
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:210]:
```suggestion
### 自定义模板[试验特性]{custom-templates-[experimental]}
```

File: content/zh/docs/setup/additional-setup/sidecar-injection/index.md [hunk 6]
```diff
@@ -48,249 +51,186 @@ $ istioctl kube-inject \
...
 
-#### 更多控制：添加例外{#more-control-adding-exceptions}
-
-有些情况下用户无法控制 pod 的创建，例如，这些用户是被其他人创建的。因此他们无法在 pod 中添加 `sidecar.istio.io/inject` 注解，来明确是否安装 sidecar。
-
-考虑在部署应用程序时创建辅助 pod 作为中间步骤。例如 [OpenShift Source to Image Builds](https://docs.okd.io/latest/builds/understanding-image-builds.html#build-strategy-s2i_understanding-image-builds)，创建这样的 pod 用于构建应用程序的源代码。构建二进制工件后，应用程序 pod 就可以运行了，而用于辅助的 pod 则被丢弃。这些中间 pod 不应该有 Istio sidecar，即使策略被设置为 `enabled`，并且名称空间被正确标记为自动注入。
+{{< warning >}}
+此功能为试行功能，可随时更改或删除。
```
Comment (kebe7jun) [inline content/zh/docs/setup/additional-setup/sidecar-injection/index.md:213]:
```suggestion
此功能为实验特性功能，可能随时更改或删除。
```

**Comment by orangegzx** [comment]
已修改

**Review by kebe7jun — APPROVED** [review]
lgtm

**Pull request merged**